### PR TITLE
SW-3833 fix 1027 error

### PR DESCRIPTION
**Pull request by JFSolution**

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -207,3 +208,13 @@ jobs:
...
+
+#  remove-artifacts:
+#    runs-on: ubuntu-latest
+#    needs: mrb-docker
+#    if: always()
+#    steps:
+#      - name: Remove artifacts
+#        uses: geekyeggo/delete-artifact@v2
+#        with:
+#          name: mrbeam_plugin
```
Comment (khaledsherkawi) [inline .github/workflows/cypress.yml]:
```suggestion
```

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -40,6 +40,7 @@ jobs:
         with:
           name: mrbeam_plugin
           path: /tmp/image.tar
+          retention-days: 1
```
Comment (khaledsherkawi) [inline .github/workflows/cypress.yml]:
```suggestion
```

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 3]
```diff
@@ -70,6 +75,11 @@ def __init__(self, plugin):
         self._fan_timers = []
         self._last_command = dict(action=None, value=None)
         self._just_initialized = False
+        self._fan_not_spinning_ts = None
+        self._fan_data_missing_ts = None
+        self._fan_data_missing_reported = None
+        self._fan_data_to_old_reported = None
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py]:
```suggestion
        self._fan_data_too_old_reported = None
```

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 4]
```diff
@@ -70,6 +75,11 @@ def __init__(self, plugin):
...
+        self._fan_data_spinning_reported = None
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py]: just for naming consistency with the variables above
```suggestion
        self._fan_not_spinning_reported = None
```

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 5]
```diff
@@ -513,13 +517,116 @@ def _validate_values(self):
...
+    def _check_if_one_value_is_none(self):
+        """
+        Checks if one of the values is None and sets the _fan_data_missing_ts flag.
+
+        Returns: (bool, list): True if one of the values is None, False otherwise. And List of error messages.
+        """
+        errs = []
+        result = False
+        if self._state is None or self._rpm is None or self._dust is None:
+            result = True
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py:537]: For readability (optional):
```suggestion
        has_none_value = any(value is None for value in (self._state, self._rpm, self._dust))
        if has_none_value:
```

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 6]
```diff
@@ -513,13 +517,116 @@ def _validate_values(self):
...
+
+            if self._dust is None:
+                errs.append("dust:{}".format(self._dust))
+
+            if self._rpm is None:
+                errs.append("rpm:{}".format(self._rpm))
+        else:
+            # reset counter or timer
+            self._fan_data_missing_ts = None
+        return result, errs
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py:553]:
```suggestion
        return has_none_value, errs
```

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 7]
```diff
@@ -513,13 +517,116 @@ def _validate_values(self):
...
+
+    def _check_if_fan_is_spinning_during_a_running_job(self):
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py:555]: This method is way too long and has possible duplications. Better move the specific error reportings into separate methods and maybe some can use the same method (not sure as there are too many parameters maybe??)

**Review by khaledsherkawi — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 5 above)
Comment (JFSolution) [reply octoprint_mrbeam/iobeam/dust_manager.py:537]: I will leave it as it was

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 7 above)
Comment (JFSolution) [reply octoprint_mrbeam/iobeam/dust_manager.py:555]: the cognitive complexity is just 3, I would keep it as it is

File: octoprint_mrbeam/iobeam/dust_manager.py [hunk 8]
```diff
@@ -70,6 +75,10 @@ def __init__(self, plugin):
         self._fan_timers = []
         self._last_command = dict(action=None, value=None)
         self._just_initialized = False
+        self._fan_not_spinning_ts = None
+        self._fan_data_missing_ts = None
+        self._fan_data_missing_reported = None
+        self._fan_data_too_old_reported = None
```
Comment (khaledsherkawi) [inline octoprint_mrbeam/iobeam/dust_manager.py:81]: self._fan_not_spinning_reported is not initialized
```suggestion
        self._fan_not_spinning_ts = None
        self._fan_data_missing_ts = None
        self._fan_data_missing_reported = None
        self._fan_data_too_old_reported = None
        self._fan_not_spinning_reported= None
```

**Review by khaledsherkawi — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/iobeam/dust_manager.py (hunk 8 above)
Comment (JFSolution) [reply octoprint_mrbeam/iobeam/dust_manager.py:81]: ohh ups, good catch

**Review by khaledsherkawi — APPROVED** [review]

**Pull request merged**